chain/router/router.go

```diff
@@ -11,36 +11,38 @@ import (
 	"google.golang.org/grpc/credentials/insecure"
 	"os"
 	routerv1 "pkg.world.dev/world-engine/rift/router/v1"
+	"time"
 
 	"pkg.berachain.dev/polaris/eth/core"
 	"pkg.berachain.dev/polaris/eth/core/types"
 
 	"cosmossdk.io/log"
 )
 
-type Result struct {
-	Code    uint64
-	Message []byte
-}
-
 type Router interface {
-	// SendMessage sends the msg payload to the game shard indicated by the namespace, if such namespace exists on chain.
-	SendMessage(ctx context.Context, namespace, sender string, msgID uint64, msg []byte) (*Result, error)
+	// SendMessage queues a message to be sent to a game shard.
+	SendMessage(_ context.Context, namespace string, sender string, msgID uint64, msg []byte) error
 	// Query queries a game shard.
 	Query(ctx context.Context, request []byte, resource, namespace string) ([]byte, error)
+	// MessageResult gets the game shard transaction result that originated from an EVM tx.
+	MessageResult(_ context.Context, evmTxHash string) ([]byte, string, uint32, error)
```

Comment:
nit: `GetMessageResult` feels better since all the other function names are also actions? Up to you not a big deal I guess

Reply:
eh, i dont mind either way. i think ill just keep it for now.